com.playeveryware.eos/Runtime/Core/Config/JsonConverter/ListOfStringsToEnumConverters.cs

```diff
@@ -0,0 +1,235 @@
+
+        protected override InputStateButtonFlags FromNumberValue(JToken token)
+        {
+            return (InputStateButtonFlags)token.Value<int>();
+        }
+    }
+
+    /// <summary>
+    /// Used to allow conversion from either a list of strings, or an enum's
+    /// "ToString()" function of strings to a single enum value.
+    /// </summary>
+    internal class ListOfStringsToAuthScopeFlags : ListOfStringsToEnumConverter<AuthScopeFlags>
+    {
+        protected override AuthScopeFlags FromStringArray(JArray array)
+        {
+            return FromStringArrayWithCustomMapping(array, AuthScopeFlagsExtensions.CustomMappings);
+        }
+
+        protected override AuthScopeFlags FromNumberValue(JToken token)
+        {
+            return (AuthScopeFlags)token.Value<int>();
+        }
+    }
+
+    /// <summary>
+    /// Used to allow conversion from either a list of strings, or an enum's
+    /// "ToString()" function of strings to a single enum value.
+    /// </summary>
+    internal class ListOfStringsToPlatformFlags : ListOfStringsToEnumConverter<WrappedPlatformFlags>
+    {
+        protected override WrappedPlatformFlags FromStringArray(JArray array)
+        {
+            Dictionary<string, WrappedPlatformFlags> wrappedCustomMapping = new();
+            foreach ((string key, PlatformFlags value) in PlatformFlagsExtensions.CustomMappings)
+            {
+                wrappedCustomMapping[key] = value.Wrap();
+            }
+
+            return FromStringArrayWithCustomMapping(array, wrappedCustomMapping);
+        }
+
+        protected override WrappedPlatformFlags FromNumberValue(JToken token)
+        {
+            return (WrappedPlatformFlags)token.Value<int>();
+        }
+    }
+#endif
+
+    /// <summary>
+    /// This class provides base functionality and declares an interface for
+    /// converter classes that can accept either a JSON array of strings, a
+    /// single value, or a single string into an enum value of the type
+    /// specified by the type parameter.
+    /// </summary>
+    /// <typeparam name="T">
+    /// Indicates the enum type that conversion functionality is being
+    /// implemented for.
+    /// </typeparam>
+    internal abstract class ListOfStringsToEnumConverter<T> : JsonConverter where T : struct, Enum
+    {
+        /// <summary>
+        /// Used as a cleaner proxy for typeof(T)
+        /// </summary>
+        private readonly Type _targetType = typeof(T);
+
+        /// <summary>
+        /// Determines whether the type that the implementing class attribute is
+        /// applied to can be converted by this JsonConverter.
+        /// </summary>
+        /// <param name="objectType">
+        /// The object to convert _to_.
+        /// </param>
+        /// <returns>
+        /// True if the conversion is supported, false otherwise.
+        /// </returns>
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(T).IsEnum;
```

Comment:
Isn't T always an Enum, because of the generic type restriction on the class?

Reply:
Yes - however implementing the abstract class still requires implementation of this function, and this is the most sensical way to do that. Sort of serves as an extra check.